homeassistant/components/tuya/vacuum.py

```diff
@@ -30,11 +31,13 @@
     "chargego": STATE_DOCKED,
     "charging": STATE_DOCKED,
     "cleaning": STATE_CLEANING,
+    "cleanning": STATE_CLEANING,
```

Comment:
This seems like a device-specific issue that should be reported upstream?

Have you contacted Tuya about this issue?

Reply:
I haven't, I'm not really sure where I'm supposed to report it as the vacuum is a branded powered by tuya thing, unlikely to get any support.

Alternatively would you be likely to pass this if I set it up as fallback values (if there is nothing found in the map but the raw tuya status contains `"clean"` or `"pause"` substrings? That way I'm not polluting the map but providing some sane fall-back values?

Reply:
Note: I've updated this change with a non-device specific fallback which still fixes my issue.  Hopefully this is acceptable! 🤞

Reply:
We don't want to support non-standard devices in our integration. The reason is that there are hundred of thousands of different device variants (not even remotely kidding on that number). If we would add exceptions/handling for them all, we will end up with an unmaintainable codebase.

Therefore, please contact Tuya support instead.

Reply:
Totally understandable that we wouldn't want to have overrides for every permutation of device... but there is nothing in my implementation which targets my device type? It's just using a slightly shortened version of the same key data above, and likely to be generic enough to fix a bunch of issues.

Promise this is the last I mention it.  If you insist, I'll remove everything bar the error handler (below comment) for this PR.

Reply:
Removed

Reply:
FYI I've resolved the requested change (removed the effected line).  Hoping this can be reviewed again when you have a sec 🙏